diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/LineLength.kt

```diff
@@ -74,6 +123,161 @@ class LineLength : Rule("line-length") {
+        } else {
+            binList.add(node)
+            binList.add(node.treeParent.findChildByType(PREFIX_EXPRESSION) ?: return)
+        }
+    }
+
+    private fun dfsForProperty(node: ASTNode, binList: MutableList<ASTNode>) {
+        node.getChildren(null).forEach {
+            if (it.elementType in PROPERTY_LIST)
+                    binList.add(it)
+            dfsForProperty(it, binList)
+        }
+    }
+
+    private fun fixProperty(parent: ASTNode, configuration: LineLengthConfiguration) {
+        if (!parent.hasChildOfType(STRING_TEMPLATE)) {
+            val newParent = parent.findChildByType(PARENTHESIZED) ?: parent
+            if (newParent.hasChildOfType(BINARY_EXPRESSION)) {
+                val leftOffset = positionByOffset(newParent.findChildByType(BINARY_EXPRESSION)!!.startOffset).second
+                fixCondition(newParent, configuration, leftOffset, !parent.hasChildOfType(PARENTHESIZED))
+            }
+        } else
+            createNodes(parent, configuration)
+    }
+
+    private fun createNodes(node: ASTNode, configuration: LineLengthConfiguration) {
+        val leftOffset = positionByOffset(node.findChildByType(STRING_TEMPLATE)!!.startOffset).second
+        if (leftOffset > configuration.lineLength)
+            return
+        var text = node.findChildByType(STRING_TEMPLATE)!!.text
+        // trim to remove first and last quote
+        text = text.trimStart(text.first()).trimEnd(text.last())
+        val lastCharIndex = configuration.lineLength.toInt() - leftOffset - STRING_PART_OFFSET
+        val indexLastSpace = (text.substring(0, lastCharIndex).lastIndexOf(' '))
+        if (indexLastSpace == -1)
+            return
+        node.removeChild(node.findChildByType(STRING_TEMPLATE)!!)
+        val prevExp = CompositeElement(BINARY_EXPRESSION)
+        node.addChild(prevExp, null)
+        createStringTemplate(text.substring(0, indexLastSpace), prevExp)
+        createNodesBetweenStringTemplates(prevExp)
+        createStringTemplate(text.substring(indexLastSpace), prevExp)
+    }
+
+    private fun createStringTemplate(text: String, prevExp: CompositeElement) {
+        val litString = CompositeElement(LITERAL_STRING_TEMPLATE_ENTRY)
+        val stringTemplate = CompositeElement(STRING_TEMPLATE)
+        val closeQuote = LeafPsiElement(CLOSING_QUOTE, "\"")
+        prevExp.addChild(litString)
+        litString.addChild(LeafPsiElement(REGULAR_STRING_PART, text))
+        prevExp.addChild(stringTemplate)
+        stringTemplate.addChild(closeQuote)
+        stringTemplate.addChild(litString, closeQuote)
+        stringTemplate.addChild(LeafPsiElement(OPEN_QUOTE, "\""), litString)
+    }
+
+    private fun createNodesBetweenStringTemplates(prevExp: CompositeElement) {
+        prevExp.addChild(PsiWhiteSpaceImpl(" "))
+        val plusOperator = CompositeElement(OPERATION_REFERENCE)
+        prevExp.addChild(plusOperator)
+        plusOperator.addChild(LeafPsiElement(PLUS, "+"))
```

Comment:
Please extract logic for creation of operation reference to a small separate method in `ASTNodeUtils` (two lines of code, creation of composite element and addition of `LeafPsiElement`). This could be useful in the future.